src/main/groovy/netflix/nebula/dependency/recommender/provider/RecommendationProviderContainer.java

```diff
@@ -46,17 +48,32 @@
     public static final RecommendationStrategies OverrideTransitives = RecommendationStrategies.OverrideTransitives;
     public static final RecommendationStrategies ConflictResolved = RecommendationStrategies.ConflictResolved;
 
-    private final Action<? super RecommendationProvider> addLastAction = new Action<RecommendationProvider>() {
-        public void execute(RecommendationProvider r) {
-            RecommendationProviderContainer.super.add(r);
-        }
-    };
-
     public RecommendationProviderContainer(Project project) {
-        super(RecommendationProvider.class, null, new RecommendationProviderNamer());
+        if (GradleVersion.current().getBaseVersion().compareTo(GradleVersion.version("6.1")) >= 0) {
```

Comment:
Unfortunately, the new recommended way how to create NamedDomainObjectList is only present in Gradle 6.1. Else branch is using soon to be a deprecated constructor. The main branch is for Gradle 6.1 but since we are still mainly on 6.0 I had to use reflection to avoid compilation error when using current Gradle 6.0